Snakefile

```diff
@@ -70,7 +71,13 @@ def make_final_input(wildcards):
         # add graph and style JSON files.
         final_input.extend(expand('{out_dir}{sep}{dataset}-{algorithm_params}{sep}gs.json',out_dir=out_dir,sep=SEP,dataset=dataset_labels,algorithm_params=algorithms_with_params))
         final_input.extend(expand('{out_dir}{sep}{dataset}-{algorithm_params}{sep}gsstyle.json',out_dir=out_dir,sep=SEP,dataset=dataset_labels,algorithm_params=algorithms_with_params))
-
+    
+    if config["analysis"]["ml"]["include"]:  
+        final_input.extend(expand('{out_dir}{sep}{dataset}-{algorithm_params}{sep}pca_image.png',out_dir=out_dir,sep=SEP,dataset=dataset_labels,algorithm_params=algorithms_with_params))
+        final_input.extend(expand('{out_dir}{sep}{dataset}-{algorithm_params}{sep}pca_components.txt',out_dir=out_dir,sep=SEP,dataset=dataset_labels,algorithm_params=algorithms_with_params))
+        final_input.extend(expand('{out_dir}{sep}{dataset}-{algorithm_params}{sep}hac_image.png',out_dir=out_dir,sep=SEP,dataset=dataset_labels,algorithm_params=algorithms_with_params))
```

Comment:
Currently we generate one plot for each algorithm-parameter combination. We only need one per dataset. I believe we can edit the patterns to look more like the `pathway-summary.txt` example.

We should also decide on file filenames. Including "image" in the name may be redundant.